spyder/widgets/ipythonconsole/client.py

```diff
@@ -163,7 +165,7 @@ def stderr_file(self):
         """Filename to save kernel stderr output."""
         if self.connection_file is not None:
             stderr_file = self.kernel_id + '.stderr'
-            stderr_file = osp.join(TEMPDIR, stderr_file)
+            stderr_file = osp.join(self.stderr_dir, stderr_file)

```

Comment:
Finally, this needs to be

    if self.stderr_dir is not None:
        stderr_file = osp.join(self.stderr_dir, stderr_file)
    else:
        stderr_file = osp.join(TEMPDIR, stderr_file)